### PR TITLE
Fix for TypeError: count() in webservices when running PHP 8.x 

### PR DESCRIPTION
**Pull request by MacJoom**
The error occurs when doing webservices API  POST call to /fields/groups/contacts/contact or /fields/groups/content/articles

Pull Request for Issue # .

### Summary of Changes

Is_array test before doing count() - not needed FieldsHelper::extract returns null if there are not two parts

### Testing Instructions

Code review -

or do test the webservices test with Postman and the collection from here: 
https://github.com/alexandreelise/j4x-api-collection

### Actual result BEFORE applying this Pull Request

TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/clients/client2/web3/web/joomla-cms/administrator/components/com_fields/src/Model/GroupModel.php:247 etc.

### Expected result AFTER applying this Pull Request

No Error, Item is added 

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Review by wilsonge — CHANGES_REQUESTED** [review]
Something looks wrong here. We call count within that function (unless we return null from the early bail https://github.com/joomla/joomla-cms/blob/7c3f3ef074ddf128dddf8d17839c105622b2c072/administrator/components/com_fields/src/Helper/FieldsHelper.php#L65)

But in that case we should be checking for parts being null. because getting the 0th element on the line 244 of null also makes no sense

**Comment by MacJoom** [comment]
Yes OK - then we do not need to a count again. I was just focusing on the part throwing the error...

**Comment by alikon** [comment]
i've got a different error  with postman without your PR

PHP Warning:  Trying to access array offset on value of type null in `/joomla-cms/administrator/components/com_fields/src/Model/GroupModel.php on line 244`

image

**Comment by alikon** [comment]
btw applying the pr no more error

image

**Comment by alikon** [comment]
tested `fields/groups/contacts/contact` too

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 0110c19164f556a17aa13287cd6ebfcfdda22f51<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40610">issues.joomla.org/tracker/joomla-cms/40610</a>.</sub>

**Comment by richard67** [comment]
@wilsonge Could you check and if ok add an approval with the review button so GitHub doesn’t show the unresolved review anymore? As far as I can see @MacJoom has solved your concerns.

**Review by wilsonge — APPROVED** [review]

File: administrator/components/com_fields/src/Model/GroupModel.php [hunk 1]
```diff
@@ -240,29 +240,30 @@ protected function preprocessForm(Form $form, $data, $group = 'content')
 
         $parts = FieldsHelper::extract($this->state->get('filter.context'));
 
+        //$parts is null if there are not two parts, early return
```
Comment (wilsonge) [inline administrator/components/com_fields/src/Model/GroupModel.php]:
```suggestion
        // If we don't have a valid context then return early
```

**Comment by obuisard** [comment]
Thank you Martin @MacJoom for the PR!

**Pull request merged**